src/core/sock/sockinfo_tcp.cpp

```diff
@@ -5790,22 +5777,13 @@ void tcp_timers_collection::handle_timer_expired(void *user_data)
     auto iter = bucket.begin();
     while (iter != bucket.end()) {
         sockinfo_tcp *p_sock = *iter;
-        // Must inc iter first bacause handle_timer_expired can erase
-        // the socket that the iter points to, with delegated timers.
+        // Must increment iterator first, the socket can be erased below in case of local timers.
         iter++;
 
-        /* It is not guaranteed that the same sockinfo object is met once
-         * in this loop.
-         * So in case sockinfo object is destroyed other processing
-         * of the same object mast be ingored.
-         * TODO Check on is_cleaned() is not safe completely.
-         */
+        // TODO Trylock can miss a timer tick and we don't trigger it in unlock() anymore.
         if (!p_sock->trylock_tcp_con()) {
-            bool destroyable = false;
-            if (!p_sock->is_cleaned()) {
-                p_sock->handle_timer_expired();
-                destroyable = p_sock->is_destroyable_no_lock();
-            }
+            p_sock->handle_timer_expired();
+            bool destroyable = p_sock->is_destroyable_no_lock();
```

Comment:
AFAIR, we call unregister_socket_timer_and_delete to both remove the timer and the delete the socket. So we may run tcp timer for a socket that is pending deletion.

Reply:
We may add is_cleaned check in handle_timer_expired tor resolve this.